osu.Game/Skinning/LegacyBeatmapSkin.cs

```diff
@@ -23,6 +24,19 @@ public LegacyBeatmapSkin(BeatmapInfo beatmap, IResourceStore<byte[]> storage, IS
             Configuration.AllowDefaultComboColoursFallback = false;
         }
 
+        public override Drawable GetDrawableComponent(ISkinComponent component)
```

Comment:
I still can't get my mind around why this is in this PR. Can it be split out and explained elsewhere..?

Reply:
I don't think it can be split out, I thought moving it to `LegacyBeatmapSkin` would explain it more easily, but put it this way:

- If a game-wide `LegacySkin` does not have the resource for HUD components (which is guessed by `!this.HasFont(Score)`), then return the default lazer skin HUD components.
- If a `LegacyBeatmapSkin` does not have the resource for HUD components, then return `null` to fall back to the game-wide `LegacySkin` HUD components.

Therefore this override is needed to achieve the second point which is returning `null` instead of returning default HUD components.

And in the end, this will all go away once falling back to default legacy skin resources is supported, as then `LegacyBeatmapSkin` can return the legacy HUD components regardless of the resources, the HUD components themselves will look up for the textures and fall back to a parent skin's textures.

Not completely sure how to put all of this in an inline comment either.

Reply:
The part where you say "this will go away" is incorrect. The beatmap skin should *never* be providing skin elements unless they are explicitly skinned.

Reply:
I still see no reason this can't be split out and tested on its own.

Reply:
> The part where you say "this will go away" is incorrect. The beatmap skin should _never_ be providing skin elements unless they are explicitly skinned.

Perhaps I was thinking about this the wrong way, so if beatmap skins are enabled, and the current `WorkingBeatmap` has a `LegacyBeatmapSkin` but not skinned, it should be ignored and the current selected skin (whether it's legacy skin or modern skin) should be used instead?

Reply:
yes, that is how things work currently and should continue working into the future.

Reply:
This has been separated to #12875 now with a test case for it, standing on its own.

*...I gotta say I actually misunderstood what you meant by "splitting" as if it's the logic that should be split somewhere else, rather than splitting the change to another PR*